### PR TITLE
Spelling

### PR DESCRIPTION
**Pull request by jsoref**
Most of these changes are to tests.

* [tests/utils/stl/test/config.py](https://github.com/jsoref/STL/blob/bee3b91f7ae2a857f8256f636f9b19abaedcc3f4/tests/utils/stl/test/config.py) is a cli api change

I'm happy to drop/adjust commits.

The `visual studio` change is probably the one most likely to be controversial. I know people like being cute and using product names, but in many projects, product names are often misspelled, so it's best to avoid whitelisting misspellings of product names. (I liked the meow and or 🐈 references.)

This PR was developed using my [check-spelling](https://github.com/marketplace/actions/check-spelling) action.

Validation for the changes can be seen in the ✔ in [spell-check](https://github.com/jsoref/STL/tree/spell-check).

The check-spelling configuration is derived from the version I worked on with @DHowett-MSFT and friends for https://github.com/microsoft/terminal.

### CODE REVIEW

File: tests/std/tests/P0220R1_optional/test.cpp [hunk 1]
```diff
@@ -1497,7 +1497,7 @@ void test_with_test_type() {
     assert(T::alive == 0);
 }
 
-void test_ambigious_assign() {
+void test_ambiguous_assign() {
```
Comment (CaseyCarter) [inline tests/std/tests/P0220R1_optional/test.cpp:1500]: No change requested here, but this test for `optional` is primarily an aggregation of the libc++ tests, so this misspelling of "ambiguous" comes from LLVM. I'd appreciate it if you could also fix this [upstream](https://github.com/llvm/llvm-project) so we don't reintroduce the error the next time we update.

File: tests/std/tests/Dev11_0000000_user_defined_literals/test.cpp [hunk 2]
```diff
@@ -402,9 +402,9 @@ int main() {
         std::experimental::erase_if(str1, is_vowel{});
         assert(str1 == "ct flffy kttns");
 
-        std::string str2{"visual studio"};
-        std::experimental::erase(str2, 'u');
-        assert(str2 == "visal stdio");
+        std::string str2{"broad anticausotic"};
+        std::experimental::erase(str2, 'o');
+        assert(str2 == "brad anticaustic");
```
Comment (StephanTLavavej) [inline tests/std/tests/Dev11_0000000_user_defined_literals/test.cpp:407]: I don't object to the intent of this change, but it's being performed inconsistently - the second example differs, and results in only one letter being removed. (Also, "anticausotic" is difficult to recognize as an actual word; apparently it's an archaic synonym for "antipyretic".) The original examples were identical and removed two letters. I suggest:

```suggestion
        std::string str2{"asynchronous beat"};
        std::experimental::erase(str2, 'a');
        assert(str2 == "synchronous bet");
```

File: tests/std/tests/Dev11_0000000_user_defined_literals/test.cpp [hunk 3]
```diff
@@ -560,9 +560,9 @@ int main() {
         assert(str1 == "ct flffy kttns");
         assert(str1_removed == 5);
 
-        std::string str2{"visual studio"};
-        const auto str2_removed = std::erase(str2, 'u');
-        assert(str2 == "visal stdio");
+        std::string str2{"broad anticaustic"};
+        const auto str2_removed = std::erase(str2, 'o');
+        assert(str2 == "brad anticaustic");
```
Comment (StephanTLavavej) [inline tests/std/tests/Dev11_0000000_user_defined_literals/test.cpp:565]:
```suggestion
        std::string str2{"asynchronous beat"};
        const auto str2_removed = std::erase(str2, 'a');
        assert(str2 == "synchronous bet");
```

**Comment by StephanTLavavej** [comment]
These are really great catches, thanks!

File: tests/std/tests/P0898R3_concepts/test.cpp [hunk 4]
```diff
@@ -2352,7 +2352,7 @@ namespace test_object_concepts {
     STATIC_ASSERT(!test_semiregular<ExplicitMoveAbomination>());
     STATIC_ASSERT(!test_semiregular<ExplicitCopyAbomination>());
     STATIC_ASSERT(!test_semiregular<CopyOnlyAbomination>());
-    STATIC_ASSERT(test_semiregular<SemiregularType>()); // Who woulda guessed?
+    STATIC_ASSERT(test_semiregular<SemiregularType>()); // Who would have guessed?
```
Comment (CaseyCarter) [inline tests/std/tests/P0898R3_concepts/test.cpp:2355]: This change in tone kills the joke (well, what there is of it anyway). I'd rather revert this change or strike the comment completely.

Comment (StephanTLavavej) [reply tests/std/tests/P0898R3_concepts/test.cpp:2355]: I'd be in favor of striking the comment.

File: tests/std/tests/P0220R1_optional/test.cpp (hunk 1 above)
Comment (jsoref) [reply tests/std/tests/P0220R1_optional/test.cpp:1500]: I'll try to get to that by this weekend.

File: tests/std/tests/P0898R3_concepts/test.cpp (hunk 4 above)
Comment (jsoref) [reply tests/std/tests/P0898R3_concepts/test.cpp:2355]:
```suggestion
    STATIC_ASSERT(test_semiregular<SemiregularType>());
```

File: tests/std/tests/Dev11_0000000_user_defined_literals/test.cpp (hunk 2 above)
Comment (jsoref) [reply tests/std/tests/Dev11_0000000_user_defined_literals/test.cpp]: Thanks, I was having a hard time finding things that worked.

**Comment by jsoref** [comment]
@StephanTLavavej: If you like them, I'll create a PR to add the action.

It's really helpful.

(And you can customize the dictionary, e.g. to remove `anticausotic` -- which, yeah, wasn't a word I'd ever seen before... it was just right next to a word that didn't have an `o` which was what I was looking for at the time...)

File: tests/std/tests/Dev11_0000000_user_defined_literals/test.cpp (hunk 3 above)
Comment (jsoref) [reply tests/std/tests/Dev11_0000000_user_defined_literals/test.cpp]: Oh, oops. That was my fault, I did a bad job copy+pasting. It was supposed to be identical. I really did know what I was trying to do.

But even so, I like your word pair much better.

File: tests/std/tests/P0220R1_optional/test.cpp (hunk 1 above)
Comment (jsoref) [reply tests/std/tests/P0220R1_optional/test.cpp:1500]: Um... well... um... [llvm](https://gist.github.com/jsoref/ca0f501e19a100db694a47c0646c3d07) is going to take a while.

I'm at `2020-04-24T11:08:53.0031077Z` working to `2020-04-24T11:08:54.3126823Z`.

That might not seem like a lot, but consider that the [run](https://github.com/jsoref/llvm-project/actions/runs/86511665) took a bit over 3 hours.
Whereas STL [runs](https://github.com/jsoref/STL/actions) take 3 1/2 to 4 minutes. And the output can include ~10 lines per millisecond. -- And this isn't fixing things, it's just identifying candidates for fixing.

I might need help approaching them with a PR. If someone here is an active contributor...
Projects tend to be fairly open to PRs w/ <20 changes. But GitHub complains when there are ~300 commits (I can easily hit 500 commits for some repositories -- but, I'm already potentially over 100 commits and I'm not done w/ the `a`s).

Comment (CaseyCarter) [reply tests/std/tests/P0220R1_optional/test.cpp:1500]: I've pushed a spelling correction to LLVM for only "ambigious" to correct this local problem. Feel free to continue fighting the good fight there, but this is now unblocked.

File: tests/std/tests/VSO_0000000_type_traits/test.cpp [hunk 5]
```diff
@@ -568,7 +568,7 @@ STATIC_ASSERT(!is_pod_v<Wrap<TrivialExceptConstruct>>);
 
 // VSO-152213 "<type_traits>: is_function does not match qualified function types"
 // VSO-154500 "<type_traits>: [Feedback]std::is_function returns false for cv/ref qualified functions"
-// C++14 LWG-2196 "Specification of is_*[copy/move]_[constructible/assignable] unclear for non-referencable types"
```
[the remark this answers is not in the document]
Comment (StephanTLavavej) [reply tests/std/tests/VSO_0000000_type_traits/test.cpp:571]: The typo is indeed in the name of https://cplusplus.github.io/LWG/issue2196 but I see no reason why we shouldn't fix it here. (The Standard consistently uses the "e" spelling.)

[the remark this answers is not in the document]
Comment (AlexGuteniev) [reply tests/std/tests/VSO_0000000_type_traits/test.cpp:571]: Probably to help search by title. But anyway, issues with number should be searched by number instead. (And web search engines are tolerant to minor corrections, even if search by name is used)

**Comment by StephanTLavavej** [comment]
Thanks again for fixing these mistakes, and congrats on your first microsoft/STL commit! 😺
